### ISSUE TITLE
MeshInboundTimeout policy

### ISSUE THREAD

**Issue by lahabana** [kumahq/kuma#3274]
### Description

This is likely to be added in an existing policy (Timeout, Retry or RateLimit)

**Comment by lahabana** [comment]
This would enable adding inbound configuration for example connection pool size, timeouts.
This contrasts with Timeout which are outbound configurations.

**Comment by jpeach** [comment]
Now sure what a "bulkhead" policy means, can we rephrase the title?

**Comment by lahabana** [comment]
I think it refers to this pattern: https://docs.microsoft.com/en-us/azure/architecture/patterns/bulkhead.

**Comment by jakubdyszkiewicz** [comment]
Default timeout on inbound HTTP routes is not specified therefore it's 15s.

**Comment by lahabana** [comment]
xref: #3934

**Comment by michaelgoodrx** [comment]
Hi, I ran into an issue with the default inbound timeout settings recently, and as a result I think this feature should be prioritized.

We can reproduce the issue with Kuma and the Node.js HTTP server, but presumably it can be reproduced with any HTTP server with the proper settings. Node.js just has especially bad defaults. This affected us in production and took a long time to figure out.

Node.js has a default keep-alive timeout time of 5 seconds for idle HTTP connections. That means if a connection is completely idle for 5 seconds, Node.js will close the connection. Somehow, the way Node does this causes Envoy to sometimes not realize the inbound connection is dead, and envoy holds onto the connection for 2 hours (in Kuma/Kong Mesh) due to the default idle timeout. So Envoy retries sending a request on the dead connection, and it gets a ECONNRESET error back. Because Envoy doesn't know if the request made it through or if the connection was dead, and Envoy also doesn't know if the request is safe to retry, it just responds with 503.

We fixed it by raising the Node.js keep-alive timeout to above the envoy idle timeout of 2 hours. That way, Envoy is the one closing the connection, not Node, so Envoy should never again send requests on a connection closed by Node.

Bottom line: I am thinking I would like to set the idle timeout for the entire mesh to something very low, maybe 0. Then for services that can handle it (ie. have the proper keep-alive setting), we can turn it up. That way, I don't have to adjust every single service's keep-alive settings. So, I'd like to have a way to define that with a Kuma policy, so I can fix it more easily and tune it for each service if needed.

**Comment by johnharris85** [comment]
I think this would be possible with the new policy selectors. An operator would be able to set the idle timeout with the mesh as the target then it could be overridden if necessary by other services / subsets.

**Comment by lahabana** [comment]
This will be part: https://github.com/kumahq/kuma/issues/4739 implemetn with #5211
